tensorflow_addons/image/filters.py

```diff
@@ -204,3 +200,95 @@ def median_filter2d(
         output = tf.cast(median, image.dtype)
         output = img_utils.from_4D_image(output, original_ndims)
         return output
+
+
+@tf.function
+def gaussian_filter2d(
+    image: FloatTensorLike,
+    sigma: FloatTensorLike,
+    filter_shape: Union[List[int], Tuple[int]] = [3, 3],
+    padding: str = "REFLECT",
+    constant_values: TensorLike = 0,
+    name: Optional[str] = None,
+) -> FloatTensorLike:
+    """
+    This function is responsible for having Gaussian Blur. It takes the image as input, computes a gaussian-kernel
+    which follows normal distribution then convolves the image with the kernel. It is implemented as 2 1D convolutions.
+    Args:
+    image: A tensor of shape
+        (batch_size, height, width, channels)
+        (NHWC), (batch_size, channels, height, width)(NCHW).
+
+    sigma:A constant of type float32. It is the standard deviation of the normal distribution.
+          The more the sigma, the more the blurring effect.
+          G(x,y)=1/(2*3.14*sigma**2)e^((x**2+y**2)/2sigma**2)
+          In 1D,
+          G(x)=e^(-x**2)/2*sigma**2
+
+    filter_shape:An `integer` or `tuple`/`list` of 2 integers, specifying
+        the height and width of the 2-D gaussian filter. Can be a single integer
+        to specify the same value for all spatial dimensions.
+
+    padding:A `string`, one of "REFLECT", "CONSTANT", or "SYMMETRIC".
+        The type of padding algorithm to use, which is compatible with
+        `mode` argument in `tf.pad`. For more details, please refer to
+        https://www.tensorflow.org/api_docs/python/tf/pad.
+
+    constant_values:A `scalar`, the pad value to use in "CONSTANT"
+        padding mode.
+    name: A name for this operation (optional).
+    Returns:
+        3D or 4D 'Tensor' of same type float64.
+    Raises:
+        Value error if:
+            1). Sigma=0
+            2). passing some string other than ["REFLECT", "CONSTANT", "SYMMETRIC"] in padding.
+    """
+
+    with tf.name_scope(name or "gaussian_filter2d"):
+        if sigma == 0:
```

Comment:
typically, we also don't want sigma < 0, right? (std should be always positive, though I know it doesn't matter because we take the square on it.)